deeplearning4j-zoo/src/main/java/org/deeplearning4j/zoo/model/AlexNet.java

```diff
@@ -0,0 +1,137 @@
+package org.deeplearning4j.zoo.model;
+
+import org.deeplearning4j.nn.api.Model;
+import org.deeplearning4j.nn.api.OptimizationAlgorithm;
+import org.deeplearning4j.nn.conf.*;
+import org.deeplearning4j.nn.conf.distribution.GaussianDistribution;
+import org.deeplearning4j.nn.conf.distribution.NormalDistribution;
+import org.deeplearning4j.nn.conf.inputs.InputType;
+import org.deeplearning4j.nn.conf.layers.ConvolutionLayer;
+import org.deeplearning4j.nn.conf.layers.DenseLayer;
+import org.deeplearning4j.nn.conf.layers.OutputLayer;
+import org.deeplearning4j.nn.conf.layers.SubsamplingLayer;
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork;
+import org.deeplearning4j.nn.weights.WeightInit;
+import org.deeplearning4j.zoo.ModelMetaData;
+import org.deeplearning4j.zoo.ZooModel;
+import org.deeplearning4j.zoo.ZooType;
+import org.nd4j.linalg.activations.Activation;
+import org.nd4j.linalg.lossfunctions.LossFunctions;
+
+
+/**
+ * AlexNet
+ *
+ * Dl4j's AlexNet model interpretation based on the original paper ImageNet Classification with Deep Convolutional Neural Networks
+ * and the imagenetExample code referenced.
+ *
+ * References:
+ * http://papers.nips.cc/paper/4824-imagenet-classification-with-deep-convolutional-neural-networks.pdf
+ * https://github.com/BVLC/caffe/blob/master/models/bvlc_alexnet/train_val.prototxt
+ *
+ * Model is built in dl4j based on available functionality and notes indicate where there are gaps waiting for enhancements.
+ *
+ * Bias initialization in the paper is 1 in certain layers but 0.1 in the imagenetExample code
+ * Weight distribution uses 0.1 std for all layers in the paper but 0.005 in the dense layers in the imagenetExample code
+ *
+ */
+public class AlexNet extends ZooModel {
+
+    private int[] inputShape = new int[] {3, 224, 224};
+    private int numLabels = 1000;
+    private long seed = 42;
+    private int iterations = 1;
+    private WorkspaceMode workspaceMode;
+    private ConvolutionLayer.AlgoMode cudnnAlgoMode;
+
+    public AlexNet(int numLabels, long seed, int iterations) {
+        this(numLabels, seed, iterations, WorkspaceMode.SEPARATE);
+    }
+
+    public AlexNet(int numLabels, long seed, int iterations, WorkspaceMode workspaceMode) {
+        this.numLabels = numLabels;
+        this.seed = seed;
+        this.iterations = iterations;
+        this.workspaceMode = workspaceMode;
+        this.cudnnAlgoMode = workspaceMode == WorkspaceMode.SINGLE ? ConvolutionLayer.AlgoMode.PREFER_FASTEST
+                        : ConvolutionLayer.AlgoMode.NO_WORKSPACE;
+    }
+
+    public ZooType zooType() {
```

Comment:
Weird, I don't know why Intellij or my compiler didn't pick up on that. Java 8 setting?